cardano-api/src/Cardano/Api/ScriptData.hs

```diff
@@ -383,11 +383,14 @@ scriptDataToJsonNoSchema = conv
 
 
     -- Script data allows any value as a key, not just string as JSON does.
-    -- For simple types we just convert them to string dirctly.
+    -- For simple types we just convert them to string directly.
     -- For structured keys we render them as JSON and use that as the string.
     convKey :: ScriptData -> Text
     convKey (ScriptDataNumber n) = Text.pack (show n)
-    convKey (ScriptDataBytes bs) = bytesPrefix
+    convKey (ScriptDataBytes bs)
+      | BSC.all (\c -> Char.isAscii c && Char.isPrint c) bs
```

Comment:
Why this change?

Reply:
A map key is often a program identifier, and it's nicer to view it decoded than in hex.

Reply:
without this,
```json
{"random name": "Vincent Ford"}
```
looks
```json
{"0x72616e646f6d206e616d65": "Vincent Ford"}
```

Reply:
I followed the comment "For simple types we just convert them to string directly."

Reply:
You missed: 
> -- * JSON hex strings with \"0x\" prefix represented as CBOR byte strings
We can't render `ScriptDataBytes` as ASCII because when parsing back to `ScriptData` the prefix `0x` followed by hex indicates that we want to encode this on chain as a CBOR bytestring.

Reply:
Have a look at `doc/reference/tx-metadata.md`

Reply:
What about strings that don't start with `0x`?

Is there another tool to convert a JSON-like datum to corresponding JSON?

Reply:
So, if a txout datum input is
```js
{"random name": "Vincent Ford"}
```
then inside the tx we have
```hs
txOutDatum = ScriptDataMap [(ScriptDataBytes "random name",ScriptDataBytes "Vincent Ford")]
```

Both key and value are ScriptDataBytes, so they they both must be displayed with no schema as 0x-base-16 strings?

Reply:
So, since this contradicts tx-metadata.md, I removed such behaviour from `scriptDataToJsonNoSchema`.